cmd/kubelet/app/options/options.go

```diff
@@ -76,6 +76,12 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.TLSPrivateKeyFile, "tls-private-key-file", s.TLSPrivateKeyFile, "File containing x509 private key matching --tls-cert-file.")
 	fs.StringVar(&s.CertDirectory, "cert-dir", s.CertDirectory, "The directory where the TLS certs are located (by default /var/run/kubernetes). "+
 		"If --tls-cert-file and --tls-private-key-file are provided, this flag will be ignored.")
+	fs.StringVar(&s.BootstrapAuthToken, "bootstrap-auth-token", s.BootstrapAuthToken, ""+
```

Comment:
@liggitt Also https://github.com/kubernetes/kubernetes/pull/30094#discussion_r73762331

So ideally this whole thing will combine with the ongoing [cluster discovery proposal](https://github.com/kubernetes/kubernetes/pull/28422), where we only pass kubelet minimal flags (one to get API server addresses, and one token for the kubelet to do this csr dance, etc.), and get rid of provisioning kubeconfig.

That being said, the proposal is still on the fly, so we could use the kubeconfig for now to get the tls certs. But my concern/question is the token here is more or less only for this bootstrap purpose, while the kubeconfig potentially serves for other purposes as well.